ThreeDSecure/src/main/java/com/braintreepayments/api/threedsecure/ThreeDSecureAPI.kt

```diff
@@ -0,0 +1,75 @@
+package com.braintreepayments.api.threedsecure
+
+import com.braintreepayments.api.core.ApiClient
+import com.braintreepayments.api.core.ApiClient.Companion.versionedPath
+import com.braintreepayments.api.core.BraintreeClient
+import com.braintreepayments.api.threedsecure.ThreeDSecureParams.Companion.fromJson
+import org.json.JSONException
+import org.json.JSONObject
+
+internal class ThreeDSecureAPI(
+    private val braintreeClient: BraintreeClient
+) {
+
+    fun performLookup(
+        request: ThreeDSecureRequest, cardinalConsumerSessionId: String?,
+        callback: ThreeDSecureResultCallback
+    ) {
+        braintreeClient.sendPOST(
+            url = versionedPath(
+                "${ApiClient.PAYMENT_METHOD_ENDPOINT}/${request.nonce}/three_d_secure/lookup"
+            ),
+            data = request.build(cardinalConsumerSessionId)
+        ) { responseBody: String?, httpError: Exception? ->
+            if (responseBody != null) {
+                try {
+                    val result: ThreeDSecureParams = fromJson(responseBody)
+                    callback.onThreeDSecureResult(result, null)
+                } catch (e: JSONException) {
+                    callback.onThreeDSecureResult(null, e)
+                }
+            } else {
+                callback.onThreeDSecureResult(null, httpError)
+            }
+        }
+    }
+
+    fun authenticateCardinalJWT(
+        threeDSecureParams: ThreeDSecureParams,
+        cardinalJWT: String?,
```

Comment:
Same thing here I don't think this should be optional

Reply:
👍 I'll make this parameter non null and add a null check prior to the call site.